modules/host/default.nix

```diff
@@ -4,18 +4,41 @@
   self,
   microvm,
   netvm,
-}: {lib, ...}: {
-  imports = [
-    (import ./minimal.nix)
+}: {
+  lib,
+  config,
+  pkgs,
+  ...
+} @ args: let
+  cfg = config.ghaf.host.minification;
+in
+  with lib; {
+    imports = [microvm.nixosModules.host];
 
-    microvm.nixosModules.host
+    options.ghaf.host.minification = with lib; {
+      reduceProfile = mkEnableOption "reduced profile";
+      disableNetwork = mkEnableOption "disableing of network";
+      disableGetty = mkEnableOption "disabling of getty";
+      removeNix = mkEnableOption "nix tooling";
```

Comment:
e.g. "Removes nix tooling from the host. Disables interaction with the host /nix/store/ and host configuration."